File: rb/features.rb
# CrunApiOverview SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module CrunApiOverviewFeatures
  def self.make_feature(name)
    case name
    when "base"
      CrunApiOverviewBaseFeature.new
    when "ratelimit"
      CrunApiOverviewRatelimitFeature.new
    when "retry"
      CrunApiOverviewRetryFeature.new
    when "test"
      CrunApiOverviewTestFeature.new
    when "timeout"
      CrunApiOverviewTimeoutFeature.new
    else
      CrunApiOverviewBaseFeature.new
    end
  end
end

File: py/features.py
# CrunApiOverview SDK feature factory

from feature.base_feature import CrunApiOverviewBaseFeature
from feature.test_feature import CrunApiOverviewTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CrunApiOverviewBaseFeature(),
        "test": lambda: CrunApiOverviewTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
